go/cmd/dolt/commands/dump.go

```diff
@@ -136,22 +117,44 @@ func (cmd DumpCmd) Exec(ctx context.Context, commandStr string, args []string, d
 		return 0
 	}
 
-	for _, tbl := range tblNames {
+	switch resultFormat {
+	case "", "sql", ".sql":
+		fileName = "doltdump.sql"
+		dumpOpts, err := getDumpArgs(fileName, resultFormat)
+		fPath, err := checkAndCreateOpenDestFile(ctx, root, dEnv, force, dumpOpts, fileName)
+		if err != nil {
+			return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		}
 
-		tblOpts := newTableArgs(tbl, dumpOpts.dest)
+		for _, tbl := range tblNames {
+			tblOpts := newTableArgs(tbl, dumpOpts.dest)
 
-		mover, verr := NewDumpDataMover(ctx, root, dEnv, tblOpts, importStatsCB, filePath)
-		if verr != nil {
-			return HandleVErrAndExitCode(verr, usage)
+			mErr := dumpTable(ctx, root, dEnv, tblOpts, fPath)
+			if mErr != nil {
+				return HandleVErrAndExitCode(mErr, usage)
+			}
 		}
+	case "csv", ".csv":
+		fileName = "doltdump/"
```

Comment:
As a followup, you should make the file name and directory name configurable as command line options (in a separate PR)